### ISSUE TITLE
[Tracker] Selection List Refactor

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#11795]
### Problem

`OptionRow` component is a beast that handles all the use cases of the App where there is a list of options. This is error-prone and hard to maintain.

### Solution

Let's refactor all different list component variations into 3 new, clean components:

- Phase 1: Radio Button List
   https://github.com/Expensify/App/issues/20352 
- Phase 2: Checkbox List
   https://github.com/Expensify/App/issues/20353
- Phase 3: Simple Selection List
   https://github.com/Expensify/App/issues/20354

----

Other Related issues

- https://github.com/Expensify/App/issues/10181
- https://github.com/Expensify/App/issues/10587
- https://github.com/Expensify/App/issues/12668
- https://github.com/Expensify/App/issues/12998

Potentially updating the List button background styles as per [this Slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1672767433644409).
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c464f37c9f85723e</li>
        <li>Upwork Job ID: 1620157290858266624</li>
        <li>Last Price Increase: 2023-01-30</li>
    </ul>
</details>

**Comment by shawnborton** [comment]
Ah interesting, I was thinking the main issue is mostly because we have a list selection on the screen at the same time as a separate text area. So I wonder if a solution to this whole thing would be to rethink the invite flow where maybe the personal message and selection rows are not on the same screen?

**Comment by mountiny** [comment]
> So I wonder if a solution to this whole thing would be to rethink the invite flow where maybe the personal message and selection rows are not on the same screen?

that is also part of the problem I would say. Can we combine other inputs with the list input on the same screen? How will it behave? I think this will benefit from the discussion about the push to page navigation decisions

**Comment by JmillsExpensify** [comment]
>Ah interesting, I was thinking the main issue is mostly because we have a list selection on the screen at the same time as a separate text area. So I wonder if a solution to this whole thing would be to rethink the invite flow where maybe the personal message and selection rows are not on the same screen?

Oh haha yeah totally agree. I just got here from the most recently added issue and came to the same conclusion. Wouldn't the custom message be a selection row and push-to-page?

**Comment by mountiny** [comment]
> Wouldn't the custom message be a selection row and push-to-page?

Makes sense, this should be discovered in the pre-design. @rushatgabhane I think there is nobody to lead this yet, @isabelastisser might need to try and get some volunteer from engineers as I dont have a time at the moment to take this on :/

**Comment by JmillsExpensify** [comment]
@rushatgabhane I don't think we've done a good job of pre-designing this in #expensify-open-source, but to give you an example of a change we're thinking of making. 

Existing Implementation
<img width="408" alt="Screen Shot 2022-10-20 at 14 03 38" src="https://user-images.githubusercontent.com/1127863/197024373-8c7d7292-7621-4673-9ed7-0ec1c08eebc9.png">

Planned future implementation
![image (5)](https://user-images.githubusercontent.com/1127863/197024901-efc4dc14-0c78-42cf-952f-8692b71b9b42.png)

The tl;dr on "push to page" for forms is this:
 - All editable fields within a form are shown as an input with a right caret
 - Tapping a field pushes to a new page to edit that field or related fields (i.e. Personal address splits out to Address/Zip/State/Country)
 - The first field within the new page is in focus by default, and making a selection or saving an entry returns the user to the previous form automatically.

**Comment by trjExpensify** [comment]
It makes sense to me, though at this point, we may as well just implement account settings minus `Personal details`. We have a concept of adding a secondary login (i.e `Phone number`) today, with a button to `Resend` the validation link etc, so we would need to move that somewhere if we update to push-to-page.

**Comment by iwiznia** [comment]
> we've done pretty extensive designs internally for how to make lists/forms/dates/etc. work more consistently than they do today

IMO this is part of the polish of WAQ and this issue too.

> Update the parts of Account Settings (plus any other areas this touches) for consistency only. So we wouldn't add OldDot features like secondary logins, or even personal details. But we could still "polish" navigation and editing in this part of the app by bringing it in line with push-to-page.

Agree

**Comment by isabelastisser** [comment]
Following this discussion before I can assign an engineer.

**Comment by twisterdotcom** [comment]
What would this amount to then?

1. Move `First name` and `Last name` into their own `Display name` drawer
2. Move `Pronouns` into its own `Pronouns` drawer
3. Move `Timezone` into its own `Timezone` drawer
4. Move `Email` and `Phone` into it's own `Contact` drawer

For the latter, would we follow the existing _Contact Methods_ design, or would we keep the settings in the drawer like this?
<img width="386" alt="image" src="https://user-images.githubusercontent.com/9133401/197240309-49fb9155-be63-458e-aa11-041ee932e97e.png">

**Comment by trjExpensify** [comment]
>For the latter, would we follow the existing Contact Methods design, or would we keep the settings in the drawer like this?

I kinda' think we should follow the Contact Methods design, otherwise the `Resend` button for the phone number secondary login will be inconsistent with the rest of the updates.

Also, do we need to add a `5.` for making sure we update the `Preferences` settings page as well?

**Comment by Beamanator** [comment]
Just catching up here, I also like the idea of moving forward with some parts of Account Settings (basically everything except the "new" Personal Details pages) so we have a solid push-to-page foundation for other pages to mimic 👍 


> Also, do we need to add a `5.` for making sure we update the `Preferences` settings page as well?

Good question @trjExpensify , would we update the dropdowns to be new pages (via push to page)? The two dropdowns I'm thinking of are:
1. Priority mode
2. Language

**Comment by twisterdotcom** [comment]
>would we update the dropdowns to be new pages (via push to page)? The two dropdowns I'm thinking of are:

I think we should, yes, but I agree, we didn't go with this in the N7 doc. We should redesign that screen to use Push to page lists.

This is tantamount to doing everything except **Personal Details** really from the N7 Account Settings doc. We'll need to handle all the RBR, offline and "Add new contact method" flows to do this properly I think, otherwise, we'll only allow adding one new contact method and then remove the button?

**Comment by Beamanator** [comment]
> I think we should, yes, but I agree, we didn't go with this in the N7 doc. We should redesign that screen to use Push to page lists.

Agreed, it would be great to have all settings pages be aligned on push-to-page implementation, then go from there. We don't have to add the Preferences page changes to the N7 Account Settings doc though, right? 🙏 

> This is tantamount to doing everything except Personal Details really from the N7 Account Settings doc. We'll need to handle all the RBR, offline and "Add new contact method" flows to do this properly I think, otherwise, we'll only allow adding one new contact method and then remove the button?

Agreed, we're already doing RBR in the App so let's add it to the Contact Methods pages too. I guess if we want to do minimal stuff we coulddd just have the "Contact methods" page & "Contact method details" page (+ RBR if that errors) without the remove / set as default, right?

**Comment by twisterdotcom** [comment]
True, we could add Remove and Set as default later.

**Comment by twisterdotcom** [comment]
Asking about some of the oddities involved here: https://expensify.slack.com/archives/C01GTK53T8Q/p1666881619925549

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by kavimuru** [comment]
#12668

**Comment by JmillsExpensify** [comment]
Regarding that issue immediately above, I think we have lots of issues around optionList, the starting point of which is [this issue](https://github.com/Expensify/Expensify/issues/228943). At some point, we might want to create a tracking for that specific initiative.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by JmillsExpensify** [comment]
Not really related to anything, but I think it would be ideal if we had either BZ members @twisterdotcom or @zanyrenney assigned to this issue, since they are working on the related list improvements that are coming with Account Settings – though of course if @isabelastisser would like to stay assigned that's fine too!

**Comment by twisterdotcom** [comment]
I won't steal all the credit from Isa, but I'll self-assign too.

**Comment by Beamanator** [comment]
Noting here that the new Pronouns page just got merged! That leaves just the Timezone page(s) which @cristipaval is working on updating 👍

**Comment by JmillsExpensify** [comment]
Woo, sweet!

**Comment by cristipaval** [comment]
Update from my end: Timezone pages are ready, the thing is that I found a bug on Web-Expensify in the `UpdateAutomaticTimezone` command and I opened [a PR](https://github.com/Expensify/Web-Expensify/pull/35624) with the fix. We need it this WE PR to be deployed in order to get Timezone pages ready for review.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Unassigning myself per Jason's comment [here](https://github.com/Expensify/App/issues/11795#issuecomment-1314218107). @twisterdotcom, I haven't participated in the discussion, so take all the credit you can here! :)

**Comment by JmillsExpensify** [comment]
My two sense. This isn't really a bug as much as a UX improvements. Accordingly, I'm removing the `bug` label.

**Comment by JmillsExpensify** [comment]
Also, I added myself so that I can keep this on my radar.

**Comment by JmillsExpensify** [comment]
Updated the title because I don't think this is a priority right now, and the `monthly` label also confirms that.

**Comment by cristipaval** [comment]
Hey @JmillsExpensify, this is already in review. I took it over from Alex and pushed it forward as a WAQ thing, according to what was discussed above 1-2 months ago. [Here](https://github.com/Expensify/App/pull/13290) is the PR.

**Comment by mountiny** [comment]
I think this issue has a little bit diverted, the issue body is not talking about the same you PRs address. I think Jason meant that the OP is task is on hold (re-designing the option lists)

**Comment by JmillsExpensify** [comment]
This would be a good one to pick up again once WAQ is over and the Account Settings revamp is complete.

**Comment by JmillsExpensify** [comment]
@twisterdotcom Are you passionate about this one? I'm super short on time, but I do think that this is increasingly a good project to take on.

**Comment by twisterdotcom** [comment]
I sort of think we can close this can't we? Given we've finished this as part of Account Settings:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/9133401/214619704-b88f1cb0-266a-4c51-969d-5c555afe6ff1.png">

And we have an issue for Preferences already:
https://github.com/Expensify/App/issues/14432

Is there a remaining task to document this standardisation somewhere?

**Comment by mountiny** [comment]
@twisterdotcom I think this issue was originally more about tracking changes to lists like this 
<img width="335" alt="image" src="https://user-images.githubusercontent.com/36083550/214625781-f12e9ae9-34a4-4efa-9083-7ac08036f406.png">

Basically such list selection is used in couple places and there is not clear guideline yet about how keyboard navigation should behave for these (going up and down and selecting using Enter).

**Comment by JmillsExpensify** [comment]
I kind of think we should close this issue. We're in the process of creating testing guidelines for the app.

**Comment by cristipaval** [comment]
We have an issue which is held on this one. If we close the current one, we have to decide what solution we want for [this](https://github.com/Expensify/App/issues/12998) particular issue.

**Comment by JmillsExpensify** [comment]
For that one, should we just not scroll to the top on each selection?

**Comment by cristipaval** [comment]
@shawnborton Instead of creating another follow up issue for aligning the design for selection and option lists in the App, I think we could use this one, now that we're discussing about closing it. I think it's related, if we are fine with how they "work", we would just change the title to *Align how list selection "looks"*. What do you think?

**Comment by shawnborton** [comment]
That sounds great to me! I think this is generally where we landed with how the lists should look:
![image](https://user-images.githubusercontent.com/2319350/215518372-e19066c8-8e60-4102-a1a8-0b100fb71b91.png)

**Comment by cristipaval** [comment]
cc @Beamanator 
As you mentioned in [the PR](https://github.com/Expensify/App/pull/14591) related to Preferences screen, in the follow-up PR that I'm going to open to fix the current issue I should take care of the following, in addition to what it was [discussed](https://github.com/Expensify/App/issues/11795#issuecomment-1408817926) together with @shawnborton:

- Set prop type as `PropTypes.arrayOf(PropTypes.object)` for `optionHoveredStyle` as mentioned [here](https://github.com/Expensify/App/pull/14591#discussion_r1095859038)
- Discuss about creating a new row component for selection lists (split the current `OptionRow` component which is too big because supports all kind of rows in the App)

**Comment by JmillsExpensify** [comment]
I think this issue is still on hold, right?

**Comment by mountiny** [comment]
@cristipaval @JmillsExpensify Could we maybe do some pre-design on how this should look and export this to Callstack? Basically if we give them the desired design (which we might already have), define the accessibility requirements (I think if we building some new general component we should consider accessibility)

But I think this is ideal for experts since this is fully front end.

**Comment by cristipaval** [comment]
yes, I totally do agree that we can make this one external. Thanks @mountiny!

**Comment by ArekChr** [comment]
Hi, @mountiny do we have pre-design and accessibility requirements for this feature?

**Comment by mountiny** [comment]
@ArekChr no I dont think this was a priority so nobody has done predesign. @cristipaval are you interested in pushing this forwards?

**Comment by ArekChr** [comment]
@cristipaval I like the plan 🙌 My question is: Is a separate proposal needed for both component splitting and for the new components for OptionRow?

**Comment by Beamanator** [comment]
I like the proposal but I feel like on point 2 we can get more than just Shawn to validate the proposal? Like If it's just refactoring I don't think we 1000% need Shawn's opinion, but definitely once we start implementing it is always nice to have him available to make sure we don't have anything off by a few pixels 👍 😬 👍 

Also I'm not 100% sure we have solutions for the problems in the OP yet:

> For inputs where we select members of a group (invite members to policy), we do not have clear guidelines on how this should work across the App. What should happen when Enter is pressed, where should the focus go after. Should backspace do anything when list option is selected?

Did we discuss stuff like pressing Enter, or maybe even arrowing up / down, selecting items with tab / space, etc etc?

**Comment by shawnborton** [comment]
Heh, happy to provide design feedback as always!

**Comment by tgolen** [comment]
I like the approach! That component is a beast and this will really help it
be more maintainable.

**Comment by burczu** [comment]
Hi I'm Bartek from Callstack - expert contributor group - since @ArekChr is on a longer sick leave now, I'll take over this issue from him

**Comment by JmillsExpensify** [comment]
@burczu Just checking in. Are you actively working on this, and if so, what's the latest status and plan?

**Comment by ArekChr** [comment]
Hey, @JmillsExpensify, I'm taking over this issue from @burczu. While I was absent no progress has been made, because of priority tasks. I will provide an update soon.

**Comment by cristipaval** [comment]
@ArekChr did a great job trying to identify the different types of options in the App, from the design standpoint. He is going to add the list here and continue planning how to tackle this massive refactor

**Comment by ArekChr** [comment]
I have been diligently working on improving the app's selection lists and made significant progress in understanding and categorizing the different components that use the `SelectionList` component. 
I have identified the similarities and differences in functionality and appearance and proposed new components with narrower scopes of responsibility to replace the current `SelectionList`.

To analyze and categorize all places that use SelectionList. I've already listed the components and their purposes. Now, try to identify similarities and differences in their functionality and appearance to create categories for different types of selection lists.

# Details

1. **Chats List**
    
    Purpose: Selection List

    Component: `FlatList` with `OptionRowLHN` as row element

    File: `LGNOptionsList.js`

    <details>
    <summary>Screenshot</summary>
    
    <img width="384" alt="chats" src="https://github.com/Expensify/App/assets/24796318/03adbaf7-d28b-4117-b66c-762c76fc0607">
    
    </details>

2. **Chats List > Group Chat -> Members**
    
    Purpose: Selection List

    Component: `OptionsList`

    File: `ReportParticipantsPage.js`
   
    <details>
    <summary>Screenshot</summary>
   
    <img width="385" alt="Zrzut%20ekranu%202023-04-7%20o%2015 56 36" src="https://github.com/Expensify/App/assets/24796318/1c36fadc-8308-48f3-bd8f-e04ec8350a4d">

    </details>
        
3. **FAB (+) > New Chat** 
    
    Purpose: Selection List

    Component `OptionsSelector`

    File: `NewChatPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="new%20chat" src="https://github.com/Expensify/App/assets/24796318/6ea84354-047f-4023-8804-cc724e5cac1a">

    </details>
        
4. **FAB (+) > New Group**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`

    File: `NewChatPage.js` with `isGroupChat: true` flag.
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="401" alt="new%20group" src="https://github.com/Expensify/App/assets/24796318/3257bf2f-d220-474b-8af1-7a33d8cd3d5c">
    
    </details>
        
5. **FAB (+) > Send Money**
    
    Purpose: Selection List

    Component: IOUParticipantsPage

    File: `MoneyRequestModal.js` with prop `props.iouType === CONST.IOU.IOU_TYPE.SEND`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="400" alt="send%20money" src="https://github.com/Expensify/App/assets/24796318/d9f34d11-1f15-4f7c-ae6f-be03e2c3107e">
    
    </details>
        
6. **FAB (+) > Request Money**
    
    Purpose: Selection List

    File: `MoneyRequestModal.js` 
    
    Component:
    
    - > `IOUConfirmPage` (*this is used nowhere except the request money screen. This might be simplified*)
    - > `IOUConfirmationList` (*the same as below*)
    - > `<OptionsSelector />` - ”Core component.”
    
    `IOUParticipantsPage`
    Uses `IOUParticipantsSplit` and `IOUParticipantsRequest`, which is used only once, passed `safeAreaPaddingBottomStyle` can be removed and implemented inside of these components.
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="402" alt="request%20money" src="https://github.com/Expensify/App/assets/24796318/cd02f004-94ae-4219-abc0-c0ea198a77b1">

    </details>
        
7. **FAB (+) > Split Bill**
    
    Purpose: Multiple Checkbox Selection List (of members)

    File: `MoneyRequestModal.js`
    
    Component:
    
    > `IOUParticipantsRequest`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="399" alt="split%20bill" src="https://github.com/Expensify/App/assets/24796318/eee8d5bd-7866-47d3-802e-50ca02b05241">

    </details>
        
8. **Search**
    
    Purpose: Selection List (of channels)

    Component: `OptionsSelector`

    File: `SearchPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="search" src="https://github.com/Expensify/App/assets/24796318/7313622f-1c3c-4790-afe3-674e70dfeb10">

    </details>
        
9. **Settings > Workspaces > Select Workspace > Manage Members**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `KeyboardDissmissingFlatList`

    File: `WorkspaceMembersPage.js`

    <details>
    <summary>Screenshot</summary>
   
    <img width="383" alt="Zrzut%20ekranu%202023-04-7%20o%2015 20 14" src="https://github.com/Expensify/App/assets/24796318/f9d1bef7-7a21-4833-94e9-b82d190ae438">

    </details>
        
10. **Settings > Workspaces > Select Workspace > Manage Members > Invite**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`
    File: `WorkspaceInvitePage.js`
    

    

    
    <details>
    <summary>Screenshot</summary>
   
    <img width="378" alt="Zrzut%20ekranu%202023-04-7%20o%2015 21 44" src="https://github.com/Expensify/App/assets/24796318/737b40b5-46d4-41dc-a5a9-09d861acc649">

    </details>
        
11. **Settings > Profile > Timezone**
    
    File: `TimezoneSelectPage.js`
    
    Component: `OptionsSelector`
    
    Purpose: Selection List Simple (label only)
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="411" alt="timezone" src="https://github.com/Expensify/App/assets/24796318/215b60f8-e0bc-4643-9835-c1f9798f0a60">

    </details>
        
12. **Settings > Profile > Pronouns**
    
    File: `ReportParticipantsPage.js`
    Component: `OptionsList`
    
    Purpose: Selection List Simple (label only)
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="397" alt="pronouns" src="https://github.com/Expensify/App/assets/24796318/c4d6f92d-cbb1-4a7c-a217-a4f7e3c8597e">

    </details>
        
13. **Settings > Profile > Personal Details > Date of Birth > Calendar > Year List Picker**
    
    File: `YearPickerPage.js`
    
    Component: `OptionsSelector`
    
    Purpose: Selection List Simple (label only)
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="390" alt="year%20picker" src="https://github.com/Expensify/App/assets/24796318/4585378c-4e52-42a1-9f81-01d04d5aba72">

    </details>
        
14. **Settings > Preferences > Priority Mode**
    
    File: `PriorityModePage.js`
    
    Component: `OptionsList`
    
    Purpose: Selection List Simple (label and text)

    <details>
    <summary>Screenshot</summary>
   
    <img width="393" alt="priority%20mode" src="https://github.com/Expensify/App/assets/24796318/030c92aa-0286-4433-89c7-0593a3831560">

    
    </details>
        
15. **Settings > Preferences > Language**
    
    File: `LanguagePage.js`
    
    Component: `OptionsList`
    
    Purpose: Selection List Simple (label only)
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="398" alt="languages" src="https://github.com/Expensify/App/assets/24796318/d63c7c2d-3910-4981-9294-0bec9366ab31">
    
    </details>

**Comment by ArekChr** [comment]
I have been diligently working on improving the app's selection lists and made significant progress in understanding and categorizing the different components that use the `SelectionList` component. 
I have identified the similarities and differences in functionality and appearance and proposed new components with narrower scopes of responsibility to replace the current `SelectionList`.

To analyze and categorize all places that use SelectionList. I've already listed the components and their purposes. Now, try to identify similarities and differences in their functionality and appearance to create categories for different types of selection lists.

# Details

1. **Chats List**
    
    Purpose: Selection List

    Component: `FlatList` with `OptionRowLHN` as row element

    File: `LGNOptionsList.js`

    <details>
    <summary>Screenshot</summary>
    
    <img width="384" alt="chats" src="https://github.com/Expensify/App/assets/24796318/03adbaf7-d28b-4117-b66c-762c76fc0607">
    
    </details>

2. **Chats List > Group Chat -> Members**
    
    Purpose: Selection List

    Component: `OptionsList`

    File: `ReportParticipantsPage.js`
   
    <details>
    <summary>Screenshot</summary>
   
    <img width="385" alt="Zrzut%20ekranu%202023-04-7%20o%2015 56 36" src="https://github.com/Expensify/App/assets/24796318/1c36fadc-8308-48f3-bd8f-e04ec8350a4d">

    </details>
        
3. **FAB (+) > New Chat** 
    
    Purpose: Selection List

    Component `OptionsSelector`

    File: `NewChatPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="new%20chat" src="https://github.com/Expensify/App/assets/24796318/6ea84354-047f-4023-8804-cc724e5cac1a">

    </details>
        
4. **FAB (+) > New Group**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`

    File: `NewChatPage.js` with `isGroupChat: true` flag.
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="401" alt="new%20group" src="https://github.com/Expensify/App/assets/24796318/3257bf2f-d220-474b-8af1-7a33d8cd3d5c">
    
    </details>
        
5. **FAB (+) > Send Money**
    
    Purpose: Selection List

    Component: IOUParticipantsPage

    File: `MoneyRequestModal.js` with prop `props.iouType === CONST.IOU.IOU_TYPE.SEND`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="400" alt="send%20money" src="https://github.com/Expensify/App/assets/24796318/d9f34d11-1f15-4f7c-ae6f-be03e2c3107e">
    
    </details>
        
6. **FAB (+) > Request Money**
    
    Purpose: Selection List

    File: `MoneyRequestModal.js` 
    
    Component:
    
    - > `IOUConfirmPage` (*this is used nowhere except the request money screen. This might be simplified*)
    - > `IOUConfirmationList` (*the same as below*)
    - > `<OptionsSelector />` - ”Core component.”
    
    `IOUParticipantsPage`
    Uses `IOUParticipantsSplit` and `IOUParticipantsRequest`, which is used only once, passed `safeAreaPaddingBottomStyle` can be removed and implemented inside of these components.
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="402" alt="request%20money" src="https://github.com/Expensify/App/assets/24796318/cd02f004-94ae-4219-abc0-c0ea198a77b1">

    </details>
        
7. **FAB (+) > Split Bill**
    
    Purpose: Multiple Checkbox Selection List (of members)

    File: `MoneyRequestModal.js`
    
    Component:
    
    > `IOUParticipantsRequest`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="399" alt="split%20bill" src="https://github.com/Expensify/App/assets/24796318/eee8d5bd-7866-47d3-802e-50ca02b05241">

    </details>
        
8. **Search**
    
    Purpose: Selection List (of channels)

    Component: `OptionsSelector`

    File: `SearchPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="search" src="https://github.com/Expensify/App/assets/24796318/7313622f-1c3c-4790-afe3-674e70dfeb10">

    </details>
        
9. **Settings > Workspaces > Select Workspace > Manage Members**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `KeyboardDissmissingFlatList`

    File: `WorkspaceMembersPage.js`

    <details>
    <summary>Screenshot</summary>
   
    <img width="383" alt="Zrzut%20ekranu%202023-04-7%20o%2015 20 14" src="https://github.com/Expensify/App/assets/24796318/f9d1bef7-7a21-4833-94e9-b82d190ae438">

    </details>
        
10. **Settings > Workspaces > Select Workspace > Manage Members > Invite**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`

    File: `WorkspaceInvitePage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="378" alt="Zrzut%20ekranu%202023-04-7%20o%2015 21 44" src="https://github.com/Expensify/App/assets/24796318/737b40b5-46d4-41dc-a5a9-09d861acc649">

    </details>
        
11. **Settings > Profile > Timezone**
    
    Purpose: Selection List Simple (label only)

    File: `TimezoneSelectPage.js`
    
    Component: `OptionsSelector`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="411" alt="timezone" src="https://github.com/Expensify/App/assets/24796318/215b60f8-e0bc-4643-9835-c1f9798f0a60">

    </details>
        
12. **Settings > Profile > Pronouns**
    
    Purpose: Selection List Simple (label only)

    File: `ReportParticipantsPage.js`

    Component: `OptionsList`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="397" alt="pronouns" src="https://github.com/Expensify/App/assets/24796318/c4d6f92d-cbb1-4a7c-a217-a4f7e3c8597e">

    </details>
        
13. **Settings > Profile > Personal Details > Date of Birth > Calendar > Year List Picker**
    
    Purpose: Selection List Simple (label only)

    Component: `OptionsSelector`

    File: `YearPickerPage.js`
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="390" alt="year%20picker" src="https://github.com/Expensify/App/assets/24796318/4585378c-4e52-42a1-9f81-01d04d5aba72">

    </details>
        
14. **Settings > Preferences > Priority Mode**
    
    Purpose: Selection List Simple (label and text)

    Component: `OptionsList`
    File: `PriorityModePage.js`
    

    


    <details>
    <summary>Screenshot</summary>
   
    <img width="393" alt="priority%20mode" src="https://github.com/Expensify/App/assets/24796318/030c92aa-0286-4433-89c7-0593a3831560">

    
    </details>
        
15. **Settings > Preferences > Language**
    
    File: `LanguagePage.js`
    
    Component: `OptionsList`
    
    Purpose: Selection List Simple (label only)
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="398" alt="languages" src="https://github.com/Expensify/App/assets/24796318/d63c7c2d-3910-4981-9294-0bec9366ab31">
    
    </details>

**Comment by ArekChr** [comment]
I have been working on analyzing the app's selection lists and made significant progress in understanding and categorizing the different components that use the `SelectionList` component. 
I have identified the similarities and differences in functionality and appearance and proposed new components with narrower scopes of responsibility to replace the current `SelectionList`.

I've prepared a raw summary that lists the components and their purposes to track this summary. Also, I included screenshots to track the old design in one place. Now, trying to identify similarities and differences in functionality and appearance to create categories for different selection lists.

## Details

1. **Chats List**
    
    Purpose: Selection List

    Component: `FlatList` with `OptionRowLHN` as row element

    File: `LGNOptionsList.js`

    <details>
    <summary>Screenshot</summary>
    
    <img width="384" alt="chats" src="https://github.com/Expensify/App/assets/24796318/03adbaf7-d28b-4117-b66c-762c76fc0607">
    
    </details>

2. **Chats List > Group Chat -> Members**
    
    Purpose: Selection List

    Component: `OptionsList`

    File: `ReportParticipantsPage.js`
   
    <details>
    <summary>Screenshot</summary>
   
    <img width="385" alt="Zrzut%20ekranu%202023-04-7%20o%2015 56 36" src="https://github.com/Expensify/App/assets/24796318/1c36fadc-8308-48f3-bd8f-e04ec8350a4d">

    </details>
        
3. **FAB (+) > New Chat** 
    
    Purpose: Selection List

    Component `OptionsSelector`

    File: `NewChatPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="new%20chat" src="https://github.com/Expensify/App/assets/24796318/6ea84354-047f-4023-8804-cc724e5cac1a">

    </details>
        
4. **FAB (+) > New Group**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`

    File: `NewChatPage.js` with `isGroupChat: true` flag.
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="401" alt="new%20group" src="https://github.com/Expensify/App/assets/24796318/3257bf2f-d220-474b-8af1-7a33d8cd3d5c">
    
    </details>
        
5. **FAB (+) > Send Money**
    
    Purpose: Selection List

    Component: IOUParticipantsPage

    File: `MoneyRequestModal.js` with prop `props.iouType === CONST.IOU.IOU_TYPE.SEND`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="400" alt="send%20money" src="https://github.com/Expensify/App/assets/24796318/d9f34d11-1f15-4f7c-ae6f-be03e2c3107e">
    
    </details>
        
6. **FAB (+) > Request Money**
    
    Purpose: Selection List

    File: `MoneyRequestModal.js` 
    
    Component:
    
    - > `IOUConfirmPage` (*this is used nowhere except the request money screen. This might be simplified*)
    - > `IOUConfirmationList` (*the same as below*)
    - > `<OptionsSelector />` - ”Core component.”
    
    `IOUParticipantsPage`
    Uses `IOUParticipantsSplit` and `IOUParticipantsRequest`, which is used only once, passed `safeAreaPaddingBottomStyle` can be removed and implemented inside of these components.
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="402" alt="request%20money" src="https://github.com/Expensify/App/assets/24796318/cd02f004-94ae-4219-abc0-c0ea198a77b1">

    </details>
        
7. **FAB (+) > Split Bill**
    
    Purpose: Multiple Checkbox Selection List (of members)

    File: `MoneyRequestModal.js`
    
    Component:
    
    > `IOUParticipantsRequest`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="399" alt="split%20bill" src="https://github.com/Expensify/App/assets/24796318/eee8d5bd-7866-47d3-802e-50ca02b05241">

    </details>
        
8. **Search**
    
    Purpose: Selection List (of channels)

    Component: `OptionsSelector`

    File: `SearchPage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="396" alt="search" src="https://github.com/Expensify/App/assets/24796318/7313622f-1c3c-4790-afe3-674e70dfeb10">

    </details>
        
9. **Settings > Workspaces > Select Workspace > Manage Members**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `KeyboardDissmissingFlatList`

    File: `WorkspaceMembersPage.js`

    <details>
    <summary>Screenshot</summary>
   
    <img width="383" alt="Zrzut%20ekranu%202023-04-7%20o%2015 20 14" src="https://github.com/Expensify/App/assets/24796318/f9d1bef7-7a21-4833-94e9-b82d190ae438">

    </details>
        
10. **Settings > Workspaces > Select Workspace > Manage Members > Invite**
    
    Purpose: Multiple Checkbox Selection List (of members)

    Component: `OptionsSelector`

    File: `WorkspaceInvitePage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="378" alt="Zrzut%20ekranu%202023-04-7%20o%2015 21 44" src="https://github.com/Expensify/App/assets/24796318/737b40b5-46d4-41dc-a5a9-09d861acc649">

    </details>
        
11. **Settings > Profile > Timezone**
    
    Purpose: Selection List Simple (label only)

    File: `TimezoneSelectPage.js`
    
    Component: `OptionsSelector`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="411" alt="timezone" src="https://github.com/Expensify/App/assets/24796318/215b60f8-e0bc-4643-9835-c1f9798f0a60">

    </details>
        
12. **Settings > Profile > Pronouns**
    
    Purpose: Selection List Simple (label only)

    File: `ReportParticipantsPage.js`

    Component: `OptionsList`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="397" alt="pronouns" src="https://github.com/Expensify/App/assets/24796318/c4d6f92d-cbb1-4a7c-a217-a4f7e3c8597e">

    </details>
        
13. **Settings > Profile > Personal Details > Date of Birth > Calendar > Year List Picker**
    
    Purpose: Selection List Simple (label only)

    Component: `OptionsSelector`

    File: `YearPickerPage.js`
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="390" alt="year%20picker" src="https://github.com/Expensify/App/assets/24796318/4585378c-4e52-42a1-9f81-01d04d5aba72">

    </details>
        
14. **Settings > Preferences > Priority Mode**
    
    Purpose: Selection List Simple (label and text)

    Component: `OptionsList`

    File: `PriorityModePage.js`
    
    <details>
    <summary>Screenshot</summary>
   
    <img width="393" alt="priority%20mode" src="https://github.com/Expensify/App/assets/24796318/030c92aa-0286-4433-89c7-0593a3831560">
    
    </details>
        
15. **Settings > Preferences > Language**
    
    Purpose: Selection List Simple (label only)

    Component: `OptionsList`

    File: `LanguagePage.js`
        
    <details>
    <summary>Screenshot</summary>
   
    <img width="398" alt="languages" src="https://github.com/Expensify/App/assets/24796318/d63c7c2d-3910-4981-9294-0bec9366ab31">
    
    </details>

_I'm editing this comment to keep all details and summaries in one place. If you like this approach, I can post more details if you think something is worth adding._

## Summary 

1. Simple Section Lists: These lists allow users to select a single item from the list. Components in this category include:
    * Chats List
    * Chats List > Group Chat -> Members
    * FAB (+) > New Chat
    * FAB (+) > Send Money
    * FAB (+) > Request Money
    * Search
2. Selection Lists (Multiple Selection with Checkbox): These lists allow users to select multiple items using checkboxes. Components in this category include:
    * FAB (+) > New Group
    * FAB (+) > Split Bill
    * Settings > Workspaces > Select Workspace > Manage Members
    * Settings > Workspaces > Select Workspace > Manage Members > Invite
3. Selection Lists (kind of radio button list): These lists are simple selection lists with a label and, in some cases, additional text. Components in this category include:
    * Settings > Profile > Timezone
    * Settings > Profile > Pronouns
    * Settings > Profile > Personal Details > Date of Birth > Calendar > Year List Picker
    * Settings > Preferences > Priority Mode
    * Settings > Preferences > Language

At first glance, I found two lists that have the same purpose but look different. Is there a reason why this is so? Can we make one consistent design for them? Namely, I'm talking about the list in Invite and Manage Members.

## Next Steps: 

Break it down into multiple components with less responsibility.

I'm currently investigating the best approach to structure one SelectionList into multiple smaller components with less functionality. If you have some concerns or questions, I'm happy to discuss that.

**Comment by twisterdotcom** [comment]
Whoah, this is great work @ArekChr.

>At first glance, I found two lists that have the same purpose but look different. Is there a reason why this is so? Can we make one consistent design for them? Namely, I'm talking about the list in Invite and Manage Members.

We absolutely can yes.

**Comment by twisterdotcom** [comment]
@ArekChr, let us know if you have an update on this now, or if you'd like any more clarification from us on anything.

**Comment by JmillsExpensify** [comment]
Wow this is amazing! In addition to what @twisterdotcom mentioned, I think we need to be careful about touching this part of the app right now. 

> Selection Lists (Multiple Selection with Checkbox): These lists allow users to select multiple items using checkboxes. Components in this category include:
...
Settings > Workspaces > Select Workspace > Manage Members
Settings > Workspaces > Select Workspace > Manage Members > Invite

I think we need to be super careful that we will soon refactor the manage members pages as part of our product roll-out for threads. As a result, I think it's best that we don't touch these two pages above for the time being.

**Comment by twisterdotcom** [comment]
@ArekChr - do you need any help splitting this all out into individual issues, or did you want to handle internally doing these in separate PR's all associated to this issue?

**Comment by ArekChr** [comment]
Hey @twisterdotcom, Sorry for the long delay and lack of updates on this task. I've had other higher-priority items to take care of in the past two weeks.
@thiagobrez will join tomorrow, and we will work together on that refactor, We will prioritize it and update you soon.

**Comment by thiagobrez** [comment]
Hey, just a quick update: I'm reading all the discussions here and will effectively start on this tomorrow. Today I got stuck with some other issues 🙇🏻

**Comment by thiagobrez** [comment]
Hey @twisterdotcom! Arek and I just sync'd, and here's the final plan we came to:

Phase 1: Radio Button List (this one should be easier because it has less usages, so we can get the feel of it)
- Step 1: Create a new, clean component for the Radio Button List, keeping the old one untouched
- Step 2: Implement storybook for this component
- Step 3: If needed and desired, perform necessary tests
- Step 4: Replace old components for the new one

Phase 2: Checkbox List (multiple selection)
- Step 1: Create a new, clean component for the Checkbox List, keeping the old one untouched
- Step 2: Implement storybook for this component
- Step 3: If needed and desired, perform necessary tests
- Step 4: Replace old components for the new one

Phase 3: Simple Selection List
- Step 1: Create a new, clean component for the Simple Selection List, keeping the old one untouched
- Step 2: Implement storybook for this component
- Step 3: If needed and desired, perform necessary tests
- Step 4: Replace old components for the new one

> At step 3 of each phase, we propose conducting tests utilising @testing-library/react-native. In the future, these tests could be integrated with Reassure (https://github.com/Expensify/App/issues/19376), which would test regressions such as render count and render time.

> We can also take Step 4 (replacing old components for the new ones) out of every phase and create a new Phase 4 for it, at the very end, if you prefer

We thought it would be good to have a new umbrella issue to track all TODOs and PR statuses, and this issue would be tied to a new baseline branch `refactor/selection_list`. This new baseline branch will be useful so we can work in parallel and avoid needing to switch back and forth between our forks.

Then, we would send smaller PRs from sub-branches into this new baseline branch. E.g:
  - `refactor/selection_list` <-- `refactor/selection_list/radio_button_list` (Phase 1)
  - `refactor/selection_list` <-- `refactor/selection_list/checkbox_list` (Phase 2)
  - `refactor/selection_list` <-- `refactor/selection_list/simple_selection_list` (Phase 3)

And finally, after every Phase is completed, we review and merge the baseline branch (`refactor/selection_list`) as a whole, to make sure no regressions pass.

What do you think? If you agree, could you help creating the issue and baseline branch for us?

Thanks!

**Comment by cristipaval** [comment]
Hey @thiagobrez! We discussed this internally and we are happy to create an umbrella issue, but we prefer to open each PR and get it merged straight into `main`.  We think that a new baseline branch is hard to maintain and we'd rather want you to make steady progress with smaller units of work that can be early tested and validated.

**Comment by thiagobrez** [comment]
Awesome @cristipaval ! All good, whatever is best for you. So, how do you wanna go -- you create the umbrella issue and us the smaller ones?

In any case, I'll start investigating Phase 1 :) 

Thanks a bunch!

**Comment by cristipaval** [comment]
Could we update the title and description for this issue and make it an umbrella? Feel free to create the other issues and link them in the new description.

**Comment by thiagobrez** [comment]
@cristipaval I think it's a good idea, but I can't, probably because I'm not the creator of this one

**Comment by twisterdotcom** [comment]
I'm going to assign you as well @thiagobrez, and unassign @JmillsExpensify, so it's just me for any Contributor review payments and the rest is on you and @cristipaval.

Ideally, whenever this goes Overdue or before, a short update on the progress in this umbrella issue would be really helpful.

**Comment by thiagobrez** [comment]
Hey @twisterdotcom! Yeah, I was gonna post an update until EOD but I guess I can just do it right now:

Working on [Phase 1 (Radio Button List)](https://github.com/Expensify/App/issues/20352). Got a clean component working with all previous features for one use-case (Timezone selection). Now, for the other use-cases it should be simpler as the base is done, and I might just need to add a few extra props for customisation. Then, I'll create a few Storybook stories to leave it well documented.

**Comment by thiagobrez** [comment]
Another update: I have all radio (single selection) lists working! 🎉 

That is:
- Timezone list
- Pronouns list
- Year picker list
- Priority Mode list
- Language picker list

**Comment by shawnborton** [comment]
Let me know when you have screenshots ready to review! We've been discussing this internally a bit and we may want to get rid of the horizontal lines in between list options.

**Comment by thiagobrez** [comment]
Hi @shawnborton! Of course, here you go:

<details>
<summary>
Timezone list
</summary>
<video src="https://github.com/Expensify/App/assets/26878038/90d564cd-e94b-458c-8d78-90999a5324eb" />
</details>

<details>
<summary>
Priority mode list
</summary>
<video src="https://github.com/Expensify/App/assets/26878038/b84d6b1f-414d-48ac-92f4-2d826c7ad04b" />
</details>

Not posting all lists because the others I mentioned in the comment above are the same as the Timezone now. Before this, they had minor differences sometimes, such as different padding values, and now they're all standardised. I tried to keep everything mostly the same, but let me know if you see anything wrong! Also, if you have a Figma file you want me to follow, that would probably be much faster and easier

**Comment by shawnborton** [comment]
Got it. I think this is looking good, but I think we can probably remove the border below each list item - I think we're chatting about making these lists look a bit cleaner and removing that border helps.

But just a heads up we'll probably be taking a look at the overall design of these lists, so they may end up changing a bit, but in the very least it will be good to refactor this to make sure all lists use the same component for the time being.

**Comment by thiagobrez** [comment]
Perfect, done 👌🏻

**Comment by thiagobrez** [comment]
Phase 1 is ready to review, left a comment in the child issue: https://github.com/Expensify/App/issues/20352

**Comment by thiagobrez** [comment]
Phase 1 PR still waiting for review: https://github.com/Expensify/App/pull/21048

Phase 2 has good progress, left an update [here](https://github.com/Expensify/App/issues/20353#issuecomment-1604491218)

**Comment by thiagobrez** [comment]
Update:

- Phase 1 PR is in final stages of review: #21048 
- Phase 2 is almost ready. Must wait Phase 1 PR to be merged before raising this PR. Left updates in the [child issue](https://github.com/Expensify/App/issues/20353)
- Phase 3 will probably be started next week, once Phase 1 is merged and Phase 2 PR is raised

**Comment by thiagobrez** [comment]
Update:

- Phase 1 PR is merged 🎉 
- Phase 2 is almost ready. Will be raising a PR early next week
- Phase 3 will probably be started next week, once Phase 2 PR is raised

**Comment by thiagobrez** [comment]
Update:

- Phase 1 PR is merged 🎉
- Phase 2 is under review [here](https://github.com/Expensify/App/pull/22622) 🚧
- Phase 3 was started. Group members list is ready

**Comment by Santhosh-Sellavel** [comment]
@cristipaval Requesting an increase in compensation for this one. More time-consuming than the other PRs. Thanks for understanding.

cc: @twisterdotcom

**Comment by Santhosh-Sellavel** [comment]
@twisterdotcom
Thanks, I agree too. 
I will let you know after requesting payment on ND.

**Comment by Santhosh-Sellavel** [comment]
Requested on ND

**Comment by anmurali** [comment]
Approved $2k for Santhosh on ND

**Comment by Santhosh-Sellavel** [comment]
To avoid confusion, we can remove Awaiting payments & update the title. 

Now phase 2 PR review is in progress

**Comment by thiagobrez** [comment]
Update:

- Phase 1 PR is merged 🎉
- Phase 2 is in final stages of review [here](https://github.com/Expensify/App/pull/22622) 🚧
- Phase 3 is almost finished, will probably already raise the PR once Phase 2 is merged. All previously investigated lists by Arek [here](https://github.com/Expensify/App/issues/11795#issuecomment-1543877225) were already replaced, but I see some lists were missed from the initial investigation (regarding all phases), so I will be adding those to Phase 3 as well.

**Comment by thiagobrez** [comment]
Hey @cristipaval! I'd like to propose adding a fourth and final phase - Phase 4 - to get the refactor wrapped up.

In short, here's what would be accomplished in each phase:
- Phase 1 (Merged):
  - created the new list component
  - replaced all lists that had single selection

- Phase 2 (PR open):
  - improved the new list component
  - replaced all lists that had multiple selection

- Phase 3 (Finishing locally):
  - improved once again the new list component
  - replaced all lists that had NO selection marker at all
  - and replaced all missing lists (from all phases) that weren't caught in the initial investigation (enumerated [here](https://github.com/Expensify/App/issues/20354#issuecomment-1655902190))

- **Phase 4 (New)**:
  - At this point, the last remaining list to replace should be the New Group list, that was put on hold due to a design decision. See here: https://github.com/Expensify/App/issues/20353#issuecomment-1602670230. So we wait for the decision and then replace it accordingly
  - After that we should have no more usages of `OptionsSelector`, `OptionsList` and `OptionRow`, so we proceed to delete those, and related files

Let me know what you think, thanks!

**Comment by twisterdotcom** [comment]
Any update on this Phase 4 @thiagobrez?

**Comment by thiagobrez** [comment]
Hey @twisterdotcom, I was still waiting for a response

**Comment by twisterdotcom** [comment]
Ah mammamia! I missed this, sorry:
>Let me know what you think, thanks!

We are waiting on this now right? https://github.com/Expensify/App/pull/22622

**Comment by thiagobrez** [comment]
@twisterdotcom No worries! Yep, that's correct 👍🏻

**Comment by thiagobrez** [comment]
Update:

- Phase 1 PR is merged 🎉
- Phase 2 is in final stages of review [here](https://github.com/Expensify/App/pull/22622) 🚧
- Phase 3 is almost finished, will probably already raise the PR once Phase 2 is merged.

**Comment by twisterdotcom** [comment]
Apologies but I am finally going on Parental Leave. I am reassigning to another member of the team.

**Comment by flaviadefaria** [comment]
I'm going OoO for 1 week but I see this is a weekly issue so should be ok. In case this needs any some attention then please assign another BZ team member.

**Comment by roryabraham** [comment]
Random piece of feedback about the current state of `SelectionList`...

- I couldn't find any scenarios where `shouldDelayFocus` really should be `false`. It seems like either the `SelectionList` has a `TextInput` (i.e: `shouldShowTextInput` === `true`), in which case we want to focus it after any navigation transitions are complete. Or it doesn't have a `TextInput`, in which case we don't want to focus it at all (because it's not there).
- Generally we try to shy away from `setTimeout`-based solutions, because they're just a little less robust.

So the solution I came up with to auto-focus the text input on `SelectionList` is, in my opinion, a bit simpler and more robust:

```diff
diff --git a/src/components/SelectionList/BaseSelectionList.js b/src/components/SelectionList/BaseSelectionList.js
index 046b64e9e5..340aa1202c 100644
--- a/src/components/SelectionList/BaseSelectionList.js
+++ b/src/components/SelectionList/BaseSelectionList.js
@@ -1,7 +1,8 @@
-import React, {useEffect, useMemo, useRef, useState} from 'react';
+import React, {useEffect, useMemo, useRef, useState, useCallback} from 'react';
 import {View} from 'react-native';
 import _ from 'underscore';
 import lodashGet from 'lodash/get';
+import {useNavigation} from '@react-navigation/native';
 import SectionList from '../SectionList';
 import Text from '../Text';
 import styles from '../../styles/styles';
@@ -42,7 +43,6 @@ function BaseSelectionList({
     keyboardType = CONST.KEYBOARD_TYPE.DEFAULT,
     onChangeText,
     initiallyFocusedOptionKey = '',
-    shouldDelayFocus = false,
     onScroll,
     onScrollBeginDrag,
     headerMessage = '',
@@ -54,10 +54,10 @@ function BaseSelectionList({
     isKeyboardShown = false,
 }) {
     const {translate} = useLocalize();
+    const navigation = useNavigation();
     const firstLayoutRef = useRef(true);
     const listRef = useRef(null);
     const textInputRef = useRef(null);
-    const focusTimeoutRef = useRef(null);
     const shouldShowTextInput = Boolean(textInputLabel);
     const shouldShowSelectAll = Boolean(onSelectAll);
     const activeElement = useActiveElement();
@@ -266,23 +266,17 @@ function BaseSelectionList({
         );
     };
 
-    /** Focuses the text input when the component mounts. If `props.shouldDelayFocus` is true, we wait for the animation to finish */
-    useEffect(() => {
-        if (shouldShowTextInput) {
-            if (shouldDelayFocus) {
-                focusTimeoutRef.current = setTimeout(() => textInputRef.current.focus(), CONST.ANIMATED_TRANSITION);
-            } else {
+    /** Focuses the text input when the component comes into focus and the animation is finished. */
+    useEffect(
+        () =>
+            navigation.addListener('transitionEnd', () => {
+                if (!shouldShowTextInput) {
+                    return;
+                }
                 textInputRef.current.focus();
-            }
-        }
-
-        return () => {
-            if (!focusTimeoutRef.current) {
-                return;
-            }
-            clearTimeout(focusTimeoutRef.current);
-        };
-    }, [shouldDelayFocus, shouldShowTextInput]);
+            }),
+        [navigation, shouldShowTextInput],
+    );
 
     /** Selects row when pressing Enter */
     useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ENTER, selectFocusedOption, {

```

It has a few advantages:

- The `shouldDelayFocus` prop is no longer needed. One less thing to think about, huzzah! As such, bugs like [this one](https://github.com/Expensify/App/issues/25860) won't be possible. You can't forget to pass `shouldDelayFocus`, because it's not there anymore.
- Using react-navigation lifecycle events rather than `setTimeout`. Just feels a bit more robust to me
- I think the code is actually a bit simpler to read as well

**Comment by roryabraham** [comment]
After noodling about a bit in [this issue](https://github.com/Expensify/App/issues/25893), I discovered some limitations with the implementation I listed above. Instead, I implemented a [similar approach](https://github.com/Expensify/App/pull/25983) using `useFocusEffect` paired with a `setTimeout` to achieve the correct auto-focusing behavior when pushing or popping screens. I still think it's a slight simplification because it works anywhere the `SelectionList` component is used and eliminates the `shouldDelayFocus` prop, which does not appear to be necessary for the same reasons listed in my previous comment.

I'm going to be mostly OOO on Monday, but since [this](https://github.com/Expensify/App/issues/25893) is a regression from this project can someone please take over my [draft PR](https://github.com/Expensify/App/pull/25983) to polish up this edge case?

**Comment by thiagobrez** [comment]
Hey @roryabraham! Thanks for the feedback and for providing a draft implementation, that is very helpful. I totally agree with what you said and will be continuing on that probably tomorrow. As a bit of context, the current implementation was just transported from the old component, as I wasn't exactly sure what was it's purpose (before those screen animation bugs started appearing). We discussed it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1692307751392999), and I was going to think of an improvement for this in Phase 3 of the refactor, but you just made it easier. Thanks!

**Comment by cristipaval** [comment]
Good collaboration there! 🚀 
Thanks @roryabraham!

**Comment by thiagobrez** [comment]
>After noodling about a bit in https://github.com/Expensify/App/issues/25893, I discovered some limitations with the implementation I listed above

@roryabraham That `transitionEnd` listener was gonna be my first option too, would you mind sharing what were the limitations with it that made you go for `useFocusEffect`?

**Comment by roryabraham** [comment]
@thiagobrez answered your question [here](https://github.com/Expensify/App/pull/26415#issuecomment-1702989957)

**Comment by thiagobrez** [comment]
Update - Phase 3 PRs:

[Reviewing] Selection highlight: https://github.com/Expensify/App/pull/27246
[Reviewing] Base implementation: https://github.com/Expensify/App/pull/27767
[Reviewing] Workspace Settings - Currency selector push-to-page: https://github.com/Expensify/App/pull/27861

**Comment by thiagobrez** [comment]
Heads up: I'll be OOO from tomorrow until Tuesday - if anything urgent comes up, someone from Callstack can pick it up!

**Comment by mollfpr** [comment]
Hi ✋ asking for clarification for https://github.com/Expensify/App/issues/21392

Will the issue be solved in this refactor?

**Comment by thiagobrez** [comment]
Hey @mollfpr, not planned - Split Bill is not part of the refactor

**Comment by thiagobrez** [comment]
Leaving updates in the Phase 3 Issue: https://github.com/Expensify/App/issues/20354#issuecomment-1739536669

**Comment by flaviadefaria** [comment]
@twisterdotcom I'm officially no longer in the BugZero team so reassigning this as it will not close until tomorrow.

**Comment by cristipaval** [comment]
Thanks @lukemorawski!

**Comment by lukemorawski** [comment]
Working on a PR

**Comment by lukemorawski** [comment]
Should have a PR tomorrow

**Comment by twisterdotcom** [comment]
@lukemorawski did you get a PR for this going?

**Comment by lukemorawski** [comment]
@twisterdotcom @dannymcclain @cristipaval @Santhosh-Sellavel Left a comment [here](https://github.com/Expensify/App/issues/20354#issuecomment-1836203311)

**Comment by lukemorawski** [comment]
chilax @MelvinBot, the issue is being discussed in a different thread

**Comment by roryabraham** [comment]
BaseOptionsSelector just got migrated from class => functional component. There's a number of active bugs / regressions with it, and we're feeling the need for some cleanup: https://expensify.slack.com/archives/C01GTK53T8Q/p1708995608771049

**Comment by roryabraham** [comment]
I think we should work to get this over the finish line. `BaseOptionsSelector` is a very large, complex component to leave in place deprecated.

**Comment by twisterdotcom** [comment]
Who is working on this now?

**Comment by roryabraham** [comment]
I'm not sure when it happened, but `BaseOptionsSelector` has been removed, so I think this is done
